query/src/main/java/com/connexta/replication/query/QueryServiceTools.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.query;
+
+import com.connexta.replication.api.NodeAdapter;
+import com.connexta.replication.api.data.Filter;
+import com.connexta.replication.api.data.FilterIndex;
+import com.connexta.replication.api.data.Site;
+import com.connexta.replication.api.impl.NodeAdapters;
+import com.connexta.replication.api.persistence.FilterIndexManager;
+import com.connexta.replication.api.persistence.FilterManager;
+import com.connexta.replication.api.persistence.SiteManager;
+import com.connexta.replication.api.queue.QueueBroker;
+import com.connexta.replication.api.queue.SiteQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+/**
+ * A facade for the services/properties that the {@link QueryService} and {@link QueryManager} need
+ * to do their jobs. This makes it easy for the QueryManager to share this functionality with each
+ * of the QueryServices it creates.
+ */
+public class QueryServiceTools {
+
+  private final NodeAdapters nodeAdapterFactories;
+
+  private final SiteManager siteManager;
+
+  private final FilterManager filterManager;
+
+  private final FilterIndexManager filterIndexManager;
+
+  private final QueueBroker queueBroker;
+
+  private final long period;
+
+  /** Default polling period for configuration changes in seconds. */
+  private static final long DEFAULT_REPLICATION_PERIOD = TimeUnit.MINUTES.toSeconds(5);
+
+  /**
+   * Creates a new instance of QueryServiceTools.
+   *
+   * @param nodeAdapterFactories An service which provides a way to retrieve {@link
+   *     com.connexta.replication.api.NodeAdapterFactory}s
+   * @param siteManager A service used to access site configurations
+   * @param filterManager A service used to access filter configurations
+   * @param filterIndexManager A service used to access indexes associated with filter
+   *     configurations
+   * @param queueBroker A service used to retrieve Queues
+   * @param period The length between polling intervals in seconds.
+   */
+  public QueryServiceTools(
+      NodeAdapters nodeAdapterFactories,
+      SiteManager siteManager,
+      FilterManager filterManager,
+      FilterIndexManager filterIndexManager,
+      QueueBroker queueBroker,
+      long period) {
```

Comment:
Not sure I understand what you're asking but we now have another application property that will set the period between query service refresh. That way there is a difference between the QueryService refresh period and the period at which each query service queries its site.

Reply:
To me, I see we have 2 configuration parameters:
- config polling period (what you have in the tools right now)
- default site polling period (what you currently pass out of band to the manager)

I am suggesting that all config be managed via your tools class as opposed to have some of the config out of band like it is today.

Reply:
hmm, I created QueryServiceTools to hold the services etc. that would need to be passed to both the services and the manager. Since the config polling period only pertains to the Manager I thought to leave it out. Either way this is a small change that I don't want to worry about right now.